[Bug Report] Unit tests fail on machine with GPU

**Describe the bug**
Unit tests are failing when run on a machine with a GPU, because `to_tokens` parameter `move_to_device` defaults to True, but the expected tokens are kept on the CPU.

**Code example**
```bash
git clone https://github.com/neelnanda-io/TransformerLens.git
cd TransformerLens
curl -sSL https://install.python-poetry.org | python3 -
PATH="/root/.local/bin:$PATH"
poetry run pytest tests/unit/test_tokenization_methods.py
```

Output:
```python
====================================== short test summary info =======================================
FAILED tests/unit/test_tokenization_methods.py::test_to_tokens_default - RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda...
FAILED tests/unit/test_tokenization_methods.py::test_to_tokens_without_bos - RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda...
FAILED tests/unit/test_tokenization_methods.py::test_to_tokens_device - RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cpu ...
FAILED tests/unit/test_tokenization_methods.py::test_get_token_position_not_found - RuntimeError: indices should be either on cpu or on the same device as the indexed tensor (cpu)
FAILED tests/unit/test_tokenization_methods.py::test_get_token_position_str - RuntimeError: indices should be either on cpu or on the same device as the indexed tensor (cpu)
FAILED tests/unit/test_tokenization_methods.py::test_get_token_position_str_without_bos - RuntimeError: indices should be either on cpu or on the same device as the indexed tensor (cpu)
==================================== 6 failed, 13 passed in 5.34s ====================================
```


**System Info**
* I ran this on a lambda labs instance with an A10
* Should be possible to repro on any linux instance with a GPU
* Python 3.8.10


### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/neelnanda-io/TransformerLens/issues) in the repo (**required**)

--- a/tests/unit/test_tokenization_methods.py
+++ b/tests/unit/test_tokenization_methods.py
@@ -1,4 +1,5 @@
 import pytest
+import torch
 from torch import Size, equal, tensor
 from transformers import AutoTokenizer
 
@@ -30,16 +31,21 @@ def test_to_tokens_default():
     s = "Hello, world!"
     tokens = model.to_tokens(s)
     assert equal(
-        tokens, tensor([[1, 11765, 14, 1499, 3]])
+        tokens, tensor([[1, 11765, 14, 1499, 3]]).to(model.cfg.device)
     ), "creates a tensor of tokens with BOS"
 
 
 def test_to_tokens_without_bos():
     s = "Hello, world!"
     tokens = model.to_tokens(s, prepend_bos=False)
-    assert equal(tokens, tensor([[11765, 14, 1499, 3]])), "creates a tensor without BOS"
+    assert equal(
+        tokens, tensor([[11765, 14, 1499, 3]]).to(model.cfg.device)
+    ), "creates a tensor without BOS"
 
 
+@pytest.mark.skipif(
+    torch.cuda.is_available(), reason="Test not relevant when running on GPU"
+)
 def test_to_tokens_device():
     s = "Hello, world!"
     tokens1 = model.to_tokens(s, move_to_device=False)

--- a/transformer_lens/HookedTransformer.py
+++ b/transformer_lens/HookedTransformer.py
@@ -664,7 +664,9 @@ class HookedTransformer(HookedRootModule):
         elif isinstance(single_token, torch.Tensor):
             single_token = single_token.item()
 
-        indices = torch.arange(len(tokens))[tokens == single_token]
+        indices = torch.arange(len(tokens), device=tokens.device)[
+            tokens == single_token
+        ]
         assert len(indices) > 0, f"The token does not occur in the prompt"
         if mode == "first":
             return indices[0].item()

